app/content-panel/tags/page.tsx

```diff
@@ -1,15 +1,157 @@
 'use client';
 
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+import { toast } from 'react-toastify';
 import styles from '../../../components/module-components/tags/Tags.module.scss';
-import AddTags from '../../../components/module-components/tags/AddTags';
 import TagTable from '../../../components/module-components/tags/TagTable';
+import TextInput from '../../../components/reusable-components/text-input/TextInput';
+import AddTag from '../../../components/module-components/tags/AddTag';
+import TagManagementControls from '../../../components/module-components/tags/TagManagementControls';
+import useDebounce from '../../../utils/hooks/useDebounce';
+import useGetTags from '../../../api/queries/tags/getTags';
+import useAddNewTag from '../../../api/mutations/tags/useAddNewTag';
+import useDeleteTag from '../../../api/mutations/tags/useDeleteTag';
+import useEditTag from '../../../api/mutations/tags/useEditTag';
+
+interface Tag {
+  id: string;
+  name: string;
+}
 
 const Tags = () => {
+  // MUTATIONS
+  const addNewTagMutation = useAddNewTag();
+  const deleteTagMutation = useDeleteTag();
+  const editTagMutation = useEditTag();
+
+  // STATE
+  const [showAddTag, setShowAddTag] = useState(false);
+  const [tags, setTags] = useState<Tag[]>([]);
+  const [editingTagId, setEditingTagId] = useState<string | null>(null);
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const debouncedSearchTerm = useDebounce(searchTerm, 300);
+  const { data, isLoading, refetch } = useGetTags(debouncedSearchTerm);
+
+  const validationSchema = Yup.object().shape({
+    tagName: Yup.string()
+      .required('Името за тагот е задолжително')
+      .test('unique', 'Тагот веќе постои', function (value) {
+        return !tags.some((tag) => tag.name.toLowerCase() === value?.toLowerCase().trim());
+      }),
+  });
+
+  const handleDelete = async (id: string) => {
+    try {
+      await deleteTagMutation.mutateAsync(id);
+      await refetch();
+      toast.success('Тагот беше успешно избришан.');
+    } catch (error) {
+      console.error(error);
+      await refetch();
+    }
+  };
+
+  const addTag = async (newTag: string) => {
+    const trimmedNewTag = newTag.trim();
+    if (tags.some((tag) => tag.name.toLowerCase() === trimmedNewTag.toLowerCase())) {
+      return { success: false, error: 'Тагот веќе постои.' };
+    }
+
+    try {
+      await addNewTagMutation.mutateAsync({
+        tagName: trimmedNewTag,
+      });
+      await refetch();
+
+      // If the mutation was successful, add the new tag to the local state
+      const newId = tags.length > 0 ? (parseInt(tags[tags.length - 1].id, 10) + 1).toString() : '1';
+      setTags([...tags, { id: newId, name: trimmedNewTag }]);
```

Comment:
since we implemented refetch you no longer have need to manipulate setTags like this